src/ui/components/SecondaryToolbox/react-devtools/suspense/inspectedElementCache.ts

```diff
@@ -0,0 +1,57 @@
+import { PauseId } from "@replayio/protocol";
+import { FrontendBridge } from "@replayio/react-devtools-inline";
+import { createCache } from "suspense";
+
+import { ReplayClientInterface } from "shared/client/types";
+import { reactDevToolsInjectionCache } from "ui/components/SecondaryToolbox/react-devtools/injectReactDevtoolsBackend";
+import {
+  ReplayWall,
+  StoreWithInternals,
+} from "ui/components/SecondaryToolbox/react-devtools/ReplayWall";
+import { InspectedReactElement } from "ui/components/SecondaryToolbox/react-devtools/types";
+import { createPromiseForRequest } from "ui/components/SecondaryToolbox/react-devtools/utils/createPromiseForRequest";
+
+let uidCounter = 0;
+
+export const inspectedElementCache = createCache<
+  [
+    replayClient: ReplayClientInterface,
+    bridge: FrontendBridge,
+    store: StoreWithInternals,
+    replayWall: ReplayWall,
+    pauseId: PauseId,
+    elementId: number
+  ],
+  InspectedReactElement
+>({
+  config: { immutable: true },
+  debugLabel: "DOMSearchCache",
+  getKey: ([replayClient, bridge, store, replayWall, pauseId, elementId]) =>
+    `${pauseId}:${elementId}`,
+  load: async ([replayClient, bridge, store, replayWall, pauseId, elementId]) => {
+    const rendererID = store.getRendererIDForElement(elementId);
+    const requestID = ++uidCounter;
+
+    // Wait until the backend has been injected before sending a message through the wall/bridge
+    await reactDevToolsInjectionCache.readAsync(replayClient, pauseId);
+    await Promise.resolve();
```

Comment:
:question: Why the extra tick here?

Reply:
Oh, something I added while debugging the backend issues I think. Meant to remove it. Good call.